geode-wan/src/test/java/org/apache/geode/management/internal/cli/functions/WanCopyRegionFunctionDelegateTest.java

```diff
@@ -0,0 +1,582 @@
+    connectionMock = mock(PooledConnection.class);
+    dispatcherMock = mock(GatewaySenderEventDispatcher.class);
+    eventCreatorMock = mock(EventCreator.class);
+    AbstractGatewaySenderEventProcessor eventProcessorMock =
+        mock(AbstractGatewaySenderEventProcessor.class);
+    RegionAttributes<?, ?> attributesMock = mock(RegionAttributes.class);
+    Set<?> idsMock = mock(Set.class);
+    internalCacheMock = mock(InternalCache.class);
+
+    function = new WanCopyRegionFunctionDelegate(clockMock, threadSleeperMock, eventCreatorMock);
+
+    doNothing().when(threadSleeperMock).sleep(anyLong());
+    when(gatewaySenderMock.getId()).thenReturn("mySender");
+    when(connectionMock.getWanSiteVersion()).thenReturn(KnownVersion.GEODE_1_15_0.ordinal());
+    when(eventCreatorMock.createGatewaySenderEvent(any(), any(), any(), any()))
+        .thenReturn(uncheckedCast(mock(GatewayQueueEvent.class)));
+    when(eventProcessorMock.getDispatcher()).thenReturn(dispatcherMock);
+    when(((AbstractGatewaySender) gatewaySenderMock).getEventProcessor())
+        .thenReturn(eventProcessorMock);
+    when(idsMock.contains(anyString())).thenReturn(true);
+    when(attributesMock.getGatewaySenderIds()).thenReturn(uncheckedCast(idsMock));
+    when(regionMock.getAttributes()).thenReturn(uncheckedCast(attributesMock));
+    when(internalCacheMock.getRegion(any())).thenReturn(uncheckedCast(regionMock));
+  }
+
+  @Test
+  public void doPostSendBatchActions_DoNotSleepIfMaxRateIsZero()
+      throws InterruptedException {
+    // arrange
+    when(internalCacheMock.getRegion(any())).thenReturn(null);
+
+    // act
+    function.wanCopyRegion(internalCacheMock, "member1", regionMock, gatewaySenderMock, 0L, 10);
+
+    // assert
+    verify(threadSleeperMock, never()).sleep(anyLong());
+  }
+
+  @Test
+  public void doPostSendBatchActions_SleepIfMaxRateIsNotZero()
+      throws InterruptedException, BatchException70 {
+    // arrange
+    Object[] options = new Object[] {"myRegion", "mySender", false, 1L, 2};
+    WanCopyRegionFunctionDelegate.ThreadSleeper sleeperMock =
+        mock(WanCopyRegionFunctionDelegate.ThreadSleeper.class);
+    doNothing().when(sleeperMock).sleep(anyLong());
+
+    // act
+    executeWanCopyRegionFunction(options, sleeperMock);
+
+    // assert
+    ArgumentCaptor<Long> captor = ArgumentCaptor.forClass(Long.class);
+    verify(sleeperMock, times(1)).sleep(captor.capture());
+    assertThat(captor.getValue()).isGreaterThan(0).isLessThanOrEqualTo(2000);
+  }
+
+  @Test
+  public void doPostSendBatchActions_ThrowInterruptedIfInterruptedTimeToSleepNotZero()
+      throws InterruptedException {
+    long maxRate = 1;
+    long elapsedTime = 20L;
+
+    // arrange
+    when(clockMock.millis()).thenReturn(startTime + elapsedTime);
+    // Sleep so that the sleep can be interrupted
+    doAnswer(invocation -> {
+      Thread.sleep(1);
+      return null;
+    }).when(threadSleeperMock).sleep(anyLong());
```

Comment:
This test is actually validating that Thread.sleep throws InterruptedException when interrupted, but you don't need (or want) to test the Java library, just your own classes. So this would be a place where it's better to just stub threadSleeperMock to throw InterruptedException:
```
  @Test
  public void doPostSendBatchActions_ThrowInterruptedIfInterruptedTimeToSleepNotZero()
      throws InterruptedException {
    // arrange
    long maxRate = 1;
    long elapsedTime = 20L;
    InterruptedException thrownByThreadSleeper = new InterruptedException("thrownByThreadSleeper");
    when(clockMock.millis()).thenReturn(startTime + elapsedTime);
    doThrow(thrownByThreadSleeper).when(threadSleeperMock).sleep(anyLong());

    // act
    Throwable thrown =
        catchThrowable(() -> function.doPostSendBatchActions(startTime, entries, maxRate));

    // assert
    assertThat(thrown).isInstanceOf(InterruptedException.class);
  }
```